stdlib/experimental/unpivot.go

```diff
@@ -146,22 +147,34 @@ func (t *unpivotTransformation) Process(chunk table.Chunk, d *execute.TransportD
 		// other_col_n
 		// _value
 
+		defaultValueColLen := 1
+		defaultFieldColLen := 1
+		if execute.HasCol(influxdb.DefaultFieldColLabel, otherCols) {
+			defaultFieldColLen = 0
+		}
+
 		groupKey := chunk.Key()
-		nCols := len(groupKey.Cols()) + len(otherCols) + 2
+		nCols := len(groupKey.Cols()) + len(otherCols) + defaultValueColLen + defaultFieldColLen
 
 		columns := make([]flux.ColMeta, 0, nCols)
 		columns = append(columns, groupKey.Cols()...)
-		columns = append(columns, flux.ColMeta{Label: "_field", Type: flux.TString})
+		if defaultFieldColLen != 0 {
+			columns = append(columns, flux.ColMeta{Label: influxdb.DefaultFieldColLabel, Type: flux.TString})
+		}
```

Comment:
This logic doesn't seem quite right to me. If there is no `_field` column in the input, then the `_field` column will never get created. We want to create it regardless, we just want to omit the column from `groupCols` and `groupValues` below if `_field` is in the `otherCols` parameter.